lib/engine/parser/split_treatment.rb

```diff
@@ -2,20 +2,38 @@ module SplitIoClient
   module Engine
     module Parser
       class SplitTreatment
-        def initialize(segments_repository)
+        def initialize(segments_repository, splits_repository, multiple = false)
+          @splits_repository = splits_repository
           @segments_repository = segments_repository
+          @multiple = multiple
+          @cache = {}
         end
 
         def call(keys, split, attributes = nil)
+          # DependencyMatcher here, split is actually a split_name in this case
+          cache_result = split.is_a? String
+          split = @splits_repository.get_split(split) if cache_result
+          digest = Digest::MD5.hexdigest("#{{matching_key: keys[:matching_key]}}#{split}#{attributes}")
```

Comment:
Ok, then if you'll have results showing not so much people are actually make use of this cache I'll get rid of it, no problem